allennlp_rc/dataset_readers/transformer_squad.py

```diff
@@ -149,14 +150,16 @@ def make_instances(
                         tokenized_context.append(wordpiece)
                     else:
                         tokenized_context.append(
-                            wordpiece._replace(idx=wordpiece.idx + token_start)
+                            dataclasses.replace(wordpiece, idx=wordpiece.idx + token_start)
```

Comment:
Can you just assign a new value?  Does it matter?  Not sure in this context if mutating the tokens is ok or not.  This way is fine, too, just seems a bit more convoluted than it needs to be.

Reply:
`.replace` does not mutate the token. I think data classes are immutable under all circumstances. `.replace` returns a new object with that one value replaced.

Reply:
https://docs.python.org/3.7/library/dataclasses.html#dataclasses.replace

Reply:
Looks like they aren't immutable: https://docs.python.org/3.7/library/dataclasses.html#frozen-instances.  🤷‍♂

Reply:
Oh, wow. So they aren't really a replacement for `NamedTuple` at all. I guess it's very pythonic though. In that case, might as well lean into it: eff92fd